services/venice-controller/src/main/java/com/linkedin/venice/controller/server/CreateVersion.java

```diff
@@ -482,24 +483,23 @@ public Route requestTopicForPushing(Admin admin) {
    * When partners have ACL issues for their push, we should provide an accurate and informative messages that
    * help partners to unblock by themselves.
    */
-  private String buildAclErrorResponse(
+  private void buildStoreAclErrorAndThrowException(
       Request request,
       Response response,
       boolean missingWriteAccess,
       boolean missingReadAccess) throws JsonProcessingException {
     response.status(HttpStatus.SC_FORBIDDEN);
-    VersionCreationResponse responseObject = new VersionCreationResponse();
     String userId = getPrincipalId(request);
     String errorMessage = "Missing [%s] ACLs for user \"" + userId + "\". Please setup ACLs for your store.";
     if (missingWriteAccess) {
       errorMessage = String.format(errorMessage, "write");
-    }
-    if (missingReadAccess) {
+    } else if (missingReadAccess) {
       errorMessage = String.format(errorMessage, "read");
+    } else {
+      errorMessage = String.format(errorMessage, "read and write");
```

Comment:
dont see any calls with both `missingWriteAccess` and `missingReadAccess` being false.

Reply:
Technically yes; This is meant to be more of a guard to not have an empty ACL error message.
e.g., I still haven't gotten to the bottom of this error message I noticed

```
Exception type: class com.linkedin.venice.exceptions.VeniceHttpException. Detailed message: Http Status 403 - Missing [{}] ACLs for user "<redacted>". Please setup ACLs for your store.
```